File: Data_Structure/10828.py
import sys

N = int(sys.stdin.readline())

stack = []
answer = []

for i in range(0,N):
    order = sys.stdin.readline().rstrip().split()
    command = order[0]
    if command == 'pop':
        if len(stack) > 0:
            answer.append(stack.pop(len(stack)-1))
        else:
            answer.append(-1)
    elif command == 'size':
        answer.append(len(stack))
    elif command == 'empty':
        if len(stack) > 0:
            answer.append(0)
        else:
            answer.append(1)
    elif command == 'top':
        if len(stack) > 0:
             answer.append(stack[len(stack)-1])
        else:
            answer.append(-1)
    else:
        stack.append(int(order[1]))

for i in range(0,len(answer)):
    print(answer[i])

File: 10828.cpp
#include "stdc++.h"

using namespace std;

int main(void) {
	ios::sync_with_stdio(0);
	cin.tie(0);

	int N;
	cin >> N;

	stack<int> st;
	for (int i=0; i<N; i++) {
		string s;
		cin >> s;
		if (s == "push") {
			int num;
			cin >> num;
			st.push(num);
		}
		else if (s == "pop") {
			if (st.empty())
				cout << -1 << "\n";
			else {
				cout << st.top() << "\n";
				st.pop();
			}
		}
		else if (s == "size")
			cout << st.size() << "\n";
		else if (s == "empty")
			cout << (int)st.empty() << "\n";
		else {
			if (st.empty())
				cout << -1 << "\n";
			else
				cout << st.top() << "\n";
		}
	}
}